rootfs/api/models.py

```diff
@@ -881,7 +891,8 @@ def publish(self, source_version='latest'):
             source_image = "{}:{}".format(source_image, source_tag)
         # If the build has a SHA, assume it's from deis-builder and in the deis-registry already
         deis_registry = bool(self.build.sha)
-        publish_release(source_image, self.config.values, self.image, deis_registry)
+        if self.build.dockerfile:
+            publish_release(source_image, self.config.values, self.image, deis_registry)
```

Comment:
publish_release only changes the tag of the image by pulling the existing image tagging it to verison instead of git sha and pushes the image back to registry.
For buildpack type apps. the image is always slugrunner  so this step is not necessary.